dockerExtension.ts

```diff
@@ -165,26 +165,23 @@ export async function activate(ctx: vscode.ExtensionContext): Promise<void> {
     activateLanguageClient(ctx);
 }
 
-export function deactivate(): Thenable<void> {
+export async function deactivate(): Promise<void> {
     if (!client) {
         return undefined;
     }
     // perform cleanup
     Configuration.dispose();
-    return client.stop();
+    return await client.stop();
 }
 
 namespace Configuration {
 
     let configurationListener: vscode.Disposable;
 
     export function computeConfiguration(params: ConfigurationParams): vscode.WorkspaceConfiguration[] {
-        if (!params.items) {

```

Comment:
Don't quite understand what this check was doing. It's safe to remove now?

Reply:
Now it returns [] instead of undefined, which is what the signature says. Also signature says params should never be undefined (as pass to us by vscode). AFAIK this is correct.